common/common-http/src/main/java/org/ow2/proactive/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public HttpClientBuilder disableContentCompression() {
+        this.useContentCompression = false;
+        return this;
+    }
+
+    public HttpClientBuilder insecure(boolean insecure) {
+        allowAnyHostname(insecure);
+        allowAnyCertificate(insecure);
+        return this;
+    }
+
+    public HttpClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    public HttpClientBuilder setDefaultRequestConfig(RequestConfig requestConfig) {
+        this.requestConfig = requestConfig;
+        return this;
+    }
+
+    public HttpClientBuilder useSystemProperties() {
+        return useSystemProperties(true);
+    }
+
+    public HttpClientBuilder useSystemProperties(boolean useSystemProperties) {
+        this.useSystemProperties = useSystemProperties;
+        return this;
+    }
+
+    public CloseableHttpClient build() {
+        org.apache.http.impl.client.HttpClientBuilder httpClientBuilder =
+                createHttpClientBuilder();
+
+        boolean acceptAnyCertificate = false;
+        boolean acceptAnyHostname = false;
+
+        if (useSystemProperties) {
+            httpClientBuilder.useSystemProperties();
+
+            String property = System.getProperty(WebProperties.WEB_HTTPS_ALLOW_ANY_CERTIFICATE);
+            if ("true".equalsIgnoreCase(property)) {
+                acceptAnyCertificate = true;
+            }
+
+            property = System.getProperty(WebProperties.WEB_HTTPS_ALLOW_ANY_HOSTNAME);
+            if ("true".equalsIgnoreCase(property)) {
+                acceptAnyHostname = true;
+            }
+        }
+
+        if (allowAnyHostname != null) {
+            if (allowAnyHostname) {
+                acceptAnyHostname = true;
+            } else {
+                acceptAnyHostname = false;
+            }
+        }
+
+        if (allowAnyCertificate != null) {
+            if (allowAnyCertificate) {
+                acceptAnyCertificate = true;
+            } else {
+                acceptAnyCertificate = false;
+            }
+        }
+
+        if (acceptAnyCertificate) {
+            httpClientBuilder.setSslcontext(createSslContext());
+        }
+
+        if (acceptAnyHostname) {
+            httpClientBuilder.setHostnameVerifier(SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+        }
+
+        if (maxConnections > 0) {
+            httpClientBuilder.setMaxConnPerRoute(maxConnections);
+            httpClientBuilder.setMaxConnTotal(maxConnections);
+        }
+
+        if (requestConfig != null) {
+            httpClientBuilder.setDefaultRequestConfig(requestConfig);
+        }
+
+        if (!useContentCompression) {
+            httpClientBuilder.disableContentCompression();
+        }
+
+        httpClientBuilder.setUserAgent("ProActive");
+
+        return httpClientBuilder.build();
+    }
+
+    public org.apache.http.impl.client.HttpClientBuilder createHttpClientBuilder() {
```

Comment:
Yep, I will move `httpClientBuilder.setUserAgent("ProActive");` in this method so that our custom build method is only responsible for mapping fields to the internal builder.